lib/modules/dosomething/dosomething_rogue/dosomething_rogue.module

```diff
@@ -208,41 +208,82 @@ function dosomething_rogue_handle_failure($values, $response = NULL, $e = NULL,
 
   // Save fail to a db log so we can easily export.
   if ($values['type'] === 'reportback') {
-    db_insert('dosomething_rogue_failed_task_log')
-      ->fields([
-        'drupal_id' => $user->uid,
-        'campaign_id' => $values['campaign_id'],
-        'campaign_run_id' => $values['campaign_run_id'],
-        'quantity' => $values['quantity'],
-        'why_participated' => $values['why_participated'],
-        'file' => $values['file'],
-        'caption' => $values['caption'],
-        'type' => $values['type'],
-        'timestamp' => REQUEST_TIME,
-        'response_code' => (isset($response->code)) ? $response->code : NULL, // @TODO: this is currently null until there a better way to get it from Gateway
-        'response_values' => (isset($e)) ? $e->getMessage() : NULL,
-        'crop_x' => $values['crop_x'],
-        'crop_y' => $values['crop_y'],
-        'crop_width' => $values['crop_width'],
-        'crop_height' => $values['crop_height'],
-        'crop_rotate' => $values['crop_rotate'],
-        'northstar_id' => $values['northstar_id'],
-      ])
-      ->execute();
+    // Check to see if the reportback item is in the dosomething_rogue_failed_task_log (by same user, campaign run, caption, and file).
+    $previously_failed = dosomething_rogue_rb_previously_failed($values['northstar_id'], $values['campaign_run_id'], $values['caption'], $values['file']);
+
+    // If the reportback has previously failed, do not enter a new record.
+    // Instead, increment the number of tries and update the time of most recent attempt to send to Rogue.
+    if ($previously_failed) {
+      // Increment the number of tries the cron job attempted to send failed record to Rogue.
+      $incremented_tries = $previously_failed[0]->tries + 1;
+      db_update('dosomething_rogue_failed_task_log')
+        ->fields([
+          'tries' => $incremented_tries,
+          'timestamp' => REQUEST_TIME,
+        ])
+        ->condition ('northstar_id', $values['northstar_id'])
+        ->condition ('campaign_run_id', $values['campaign_run_id'])
+        ->condition ('caption', $values['caption'])
+        ->condition ('file', $values['file'])
+        ->execute();
```

Comment:
That seems like a good idea! also maybe if you broke them into their own functions you can use it when you do the inserts and updates below.